utils.py

```diff
@@ -352,3 +354,28 @@ def coco_evaluation(gt_filepath, prediction_filepath, image_ids, category_ids):
     cocoEval.summarize()
 
     return cocoEval.stats[1], cocoEval.stats[8]
+
+
+def dense_crf(img, output_probs, compat_gaussian=3, sxy_gaussian=1, compat_bilateral=10, sxy_bilateral=1, srgb=50):
+
+    h = output_probs.shape[1]
+    w = output_probs.shape[2]
+
+    d = DenseCRF2D(w, h, 2)
+    U = unary_from_softmax(output_probs)
+    org_img = denormalize_img(img, mean=[0.0], std=[1.0]) * 255.
+    org_img = org_img.transpose(1, 2, 0)
+    org_img = np.ascontiguousarray(org_img, dtype=np.uint8)
```

Comment:
@taraspiotr one thing that could be interesting is to pass a feature activation from intermediate layer instead of original image. On the other hand it seems that if one was to put work into this one could implement CRF inside of network approach.